detection-rules/impersonation_mailgun.yml

```diff
@@ -0,0 +1,54 @@
+name: "Brand Impersonation: Mailgun"
+description: "Impersonation of the Mailgun Email delivery platform."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and 2 of (
+    regex.icontains(sender.display_name, '\bmailgun\b'),
+    strings.ilike(sender.email.domain.domain, '*mailgun*'),
```

Comment:
`ilike` here can probably be a icontains? I'm not sure which is more performant, but it reads better IMO.